components/director/internal/domain/formation/service.go

```diff
@@ -149,12 +159,24 @@ func (s *service) AssignFormation(ctx context.Context, tnt, objectID string, obj
 
 // UnassignFormation unassigns object base on graphql.FormationObjectType.
 // If the graphql.FormationObjectType is graphql.FormationObjectTypeApplication it removes the provided formation from the
-// scenario label of the application. If the graphql.FormationObjectType is graphql.FormationObjectTypeTenant it will
+// scenario label of the application.
+// If the graphql.FormationObjectType is graphql.FormationObjectTypeRuntime and the provided formation is not coming from ASA,
+// it removes the formation from the scenario label of the runtime,
+// but if the provided formation is assigned from ASA it does nothing.
+// If the graphql.FormationObjectType is graphql.FormationObjectTypeTenant it will
 // delete the automatic scenario assignment with the caller and target tenant.
 func (s *service) UnassignFormation(ctx context.Context, tnt, objectID string, objectType graphql.FormationObjectType, formation model.Formation) (*model.Formation, error) {
 	switch objectType {
 	case graphql.FormationObjectTypeApplication:
-		return s.modifyAssignedFormationsForApplication(ctx, tnt, objectID, formation, deleteFormation)
+		return s.modifyAssignedFormations(ctx, tnt, objectID, formation, model.ApplicationLabelableObject, deleteFormation)
```

Comment:
Use objectTypeToLabelableObject(objectType) instead of hardcoding the objectType